ros2_batch_job/__main__.py

```diff
@@ -307,7 +308,9 @@ def process_coverage(args, job):
             '--object-directory=' + package_build_path,
             '-k',
             '-r', os.path.abspath('.'),
-            '--xml', '--output=' + outfile,
+            '--xml', '--output=' + outfile,         
+            '--gcov-exclude=.*#tests?#.*',
+            '--gcov-exclude=.*#gtest_vendor#.*',
```

Comment:
Actually something else occurred to me. Are there implicit anchor metacharacters around these expressions that make the `.*` wrappers necessary? If not couldn't these be simplified to `#tests?#` and `#gtest_vendor#` respectively?

Reply:
The gcov generated files are in the form:
 ```
install#gtest_vendor#src#gtest_vendor#include#gtest#gtest.h.gcov                    src#ros2#rcl_logging#rcl_logging_spdlog#src#rcl_logging_spdlog.cpp.gcov
install#gtest_vendor#src#gtest_vendor#include#gtest#gtest-printers.h.gcov           src#ros2#rcl_logging#rcl_logging_spdlog#test#test_logging_interface.cpp.gcov
install#gtest_vendor#src#gtest_vendor#include#gtest#internal#gtest-internal.h.gcov
```
My local testing suggest that we can not remove the `.*`:

```
jrivero@shappire coverage_ws $ gcovr --verbose --gcov-exclude='.*#gtest_vendor#.*' --object-directory=build/rcl_logging_spdlog -k -r /home/jrivero/code/ros2/coverage_ws  -g | grep Excluding
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#gtest.h.gcov
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#internal#gtest-internal.h.gcov
Excluding gcov file: /home/jrivero/code/ros2/coverage_ws/install#gtest_vendor#src#gtest_vendor#include#gtest#gtest-printers.h.gcov
jrivero@shappire coverage_ws $ gcovr --verbose --gcov-exclude='#gtest_vendor#' --object-directory=build/rcl_logging_spdlog -k -r /home/jrivero/code/ros2/coverage_ws  -g | grep Excludingjrivero@shappire coverage_ws $ 
```